### DISCUSSION TITLE
Pan and Zoom Limiting

### DISCUSSION THREAD

**Discussion by coleblarson** [Mapsui/Mapsui#2094] [General]
I just updated to 4.0 from a version (4 beta something) that allowed setting pan and zoom bounds on the Navigator.   That is no longer available in this version.

How does one do this now? 

I've tried this (and a bunch of other things):
`            
            _mapControl = mapControl;

            _map = _mapControl.Map;
            _map.Info += OnMapInfoChanged!;
           
            _map.CRS = "EPSG:3857"; // required for scale bar widget, Spherical Mercator projection
            _map.Layers.Add(Mapsui.Tiling.OpenStreetMap.CreateTileLayer());

            _latCenter = 38.0;
            _lonCenter = -119.0; 

            double extent = 1; //1 deg = 111 km, 60 nautical miles

            MPoint southWestPoint = new MPoint(_lonCenter - extent, _latCenter - extent); //south west extent
            MPoint northEastPoint = new MPoint(_lonCenter + extent, _latCenter + extent); //north east extent

            _center = SphericalMercator.FromLonLat(new MPoint(_lonCenter, _latCenter));//center point in spherical mercator projection
            _southWest = SphericalMercator.FromLonLat(southWestPoint); //south west point in spherical mercator projection
            _northEast = SphericalMercator.FromLonLat(northEastPoint); //north east point in spherical mercator projection

            double width = _northEast.X - _southWest.X;
            double height = _northEast.Y - _southWest.Y;

            _limitedViewport = new Viewport(_center.X, _center.Y, double.MaxValue, 0, width, height);           

            _map.Navigator.SetViewport(_limitedViewport);`

... to no avail

One of the other things I've tried was setting OverrridePanBounds but this produced results where the zoom widget sort of worked for a few dozen clicks but the mouse wheel completely ignored them.

I'm developing on .net core 7 in WPF

**Comment by inforithmics** [comment]
I think this sample should help
https://github.com/Mapsui/Mapsui/blob/master/Samples/Mapsui.Samples.Common/Maps/Navigation/KeepWithinExtentSample.cs
